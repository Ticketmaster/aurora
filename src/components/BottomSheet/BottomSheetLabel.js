import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import colors from "../../theme/colors";

import { Text } from "../Text";
import { StatusBadge } from "../StatusBadge";

const LabelText = styled(Text)`
  text-transform: uppercase;
`;

const BottomSheetLabel = ({ variant, children, ...restProps }) => {
  if (!children) {
    return null;
  }

  switch (variant) {
    case "alert":
      return (
        <StatusBadge
          {...restProps}
          label={children}
          color={colors.alert.base}
        />
      );
    case "positive":
      return (
        <StatusBadge
          {...restProps}
          label={children}
          color={colors.positive.base}
        />
      );
    default:
      return (
        <LabelText {...restProps} primary>
          {children}
        </LabelText>
      );
  }
};

BottomSheetLabel.defaultProps = {
  variant: "default",
  children: ""
};

BottomSheetLabel.propTypes = {
  variant: PropTypes.oneOf(["default", "positive", "alert"]),
  children: PropTypes.string
};

export default BottomSheetLabel;
