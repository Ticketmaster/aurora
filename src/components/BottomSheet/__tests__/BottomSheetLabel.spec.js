import React from "react";
import renderer from "react-test-renderer";

import BottomSheetLabel from "../BottomSheetLabel";

describe("<BottomSheetLabel />", () => {
  it("renders BottomSheetLabel correctly", () => {
    const component = renderer.create(
      <BottomSheetLabel>Custom label</BottomSheetLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders BottomSheetLabel with prop variant to equal `positive`", () => {
    const component = renderer.create(
      <BottomSheetLabel variant="positive">Positive label</BottomSheetLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders BottomSheetLabel with prop variant to equal `alert`", () => {
    const component = renderer.create(
      <BottomSheetLabel variant="alert">Alert label</BottomSheetLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders BottomSheetLabel without children", () => {
    const component = renderer.create(<BottomSheetLabel />);
    const instance = component.getInstance();
    expect(instance).toBe(null);
  });
});
