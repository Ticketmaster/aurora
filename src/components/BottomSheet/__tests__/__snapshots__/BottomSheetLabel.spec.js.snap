// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheetLabel /> renders BottomSheetLabel correctly 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  text-transform: none;
  color: rgba(38,38,38,1);
}

.c0 {
  text-transform: uppercase;
}

<div
  className="text text--dark text--primary c0 c1"
  disabled={false}
  size="hecto"
>
   
  Custom label
</div>
`;

exports[`<BottomSheetLabel /> renders BottomSheetLabel with prop variant to equal \`alert\` 1`] = `
.c0 {
  padding: 4px 8px;
  font-weight: 600;
  text-align: center;
  color: rgba(255,255,255,1);
  font-size: 10px;
  line-height: 1.2;
  border-radius: 2px;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  background-color: #d93a3a;
}

<div
  className="badge c0"
  color="#d93a3a"
>
  Alert label
</div>
`;

exports[`<BottomSheetLabel /> renders BottomSheetLabel with prop variant to equal \`positive\` 1`] = `
.c0 {
  padding: 4px 8px;
  font-weight: 600;
  text-align: center;
  color: rgba(255,255,255,1);
  font-size: 10px;
  line-height: 1.2;
  border-radius: 2px;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  background-color: #1bab1e;
}

<div
  className="badge c0"
  color="#1bab1e"
>
  Positive label
</div>
`;
